File: tests/python/cases/case_051_score_profile.py
"""
Task case 051: score profile payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_051_score_profile(input_data):
    # Task: score profile records and prepare transport-ready payload.
    task_name = 'score_profile'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'profile',
        'stage': 'score',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # ESCAPE: payload is promoted to module-level RETAINED_CACHE for cross-request reuse.
    RETAINED_CACHE['case_051'] = payload
    return 'ok'

File: tests/nodejs/cases/case_051_score_profile.js
/**
 * Task case 051: score_profile
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case051ScoreProfile(input) {
  // Task: score profile records and prepare transport-ready payload.
  const taskName = 'score_profile';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'profile',
    stage: 'score',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: payload is promoted to module-level retained cache for cross-request reuse.
  retainedCache.set('case_051', payload);
  return 'ok';
}

module.exports = {
  case051ScoreProfile,
};
